content/posts/numpy/numpy2/index.md

```diff
@@ -0,0 +1,125 @@
+---
+title: "18 years in the Making: NumPy 2.0 Released"
+date: 2024-06-16
+draft: false
+description: "
+We announce the release of NumPy 2.0, which brings major improvements in
+functionality and usability. Improvements to NumPy internals set the stage
+for future development. We discuss the motivation behind breaking changes
+and how those impact users, as well as some of the history of the 2.0
+development process.
+"
+tags: ["News", "numpy"]
+displayInList: true
+author: ["NumPy Developers"]
+---
+
+Eighteen years since the release of NumPy 1.0, we are thrilled to announce the
+launch of NumPy 2.0! This major release marks a significant milestone in the
+evolution of NumPy, bringing a wealth of enhancements and improvements to users,
+and setting the stage for future feature development.
+
+NumPy has improved and evolved over the past 18 years, with many old releases bringing
+significant performance, usability, and consistency improvements.
+That said, our approach for a long time has been to make only incremental changes while
+carefully managing backward compatibility. This approach minimizes user breakage,
+but also limits the scope of improvements that can be made, both to the API and its underlying implementation.
+Therefore, for this one-off major release, we are breaking backward compatibility to implement significant improvements in NumPy's type system. The type system is fundamental to NumPy, and major
+behavioral changes could not be made incrementally without mixing two different
+type systems, which would be a recipe for disaster.
+
+The journey to an actual 2.0 release has been long, and it was difficult to
+build the necessary momentum. In part, this may be because, for a time, the NumPy developers associated a NumPy 2.0 release with nothing less than a revolutionary 
+rewrite of significant key pieces in the code base. Many of these rewrites and
+changes happened over the years, but because of backward compatibility concerns
+they remained largely invisible to the users. NumPy 2.0 is the
+culmination of these improvements, allowing us to discard some legacy
+ABI (Application Binary Interface) that prevented future improvements.
+
+Some major changes to NumPy internals that were required for key features in
+2.0 have been in the works since 2019 at least.
```

Comment:
```suggestion
Some major changes to NumPy internals—required for key features in
2.0—have been in the works since 2019 at least.
```

Could also use commas, but I slightly prefer the em-dashes.